HashPlayer.java

```diff
@@ -1,23 +1,58 @@
-//
 import java.util.*;
-public class Player {
+
+public class HashPlayer implements PlayerInterface {
 	// Valid is Static so All Instances: Player 1 and Player 2 Share the Variable
 	static HashSet<Integer> used = new HashSet<Integer>();
+	static HashMap<Integer, String> bothMoves = new HashMap<Integer, String>();
 	static int turns = 0;
-	int no;
+	
+	int no; // Player Number

```

Comment:
Probably best to document all variables - doesn't have to be perfect but should at least document why each field is needed